File: API/Entities/Tasks.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Entities
{

    public enum TaskStatus{
        InProgress, Completed
    }

    public class Tasks
    {
        public int Id { get; set; }

        public string Title { get; set; }

        public string Description { get; set; }

        [Column(TypeName="Date")]
        public DateTime StartTime { get; set; }
        
        public int EmployeeId { get; set; }
        
        public AppUser Employee { get; set; }

        public int Duration { get; set; }

        public ICollection<SubTask> SubTasks { get; set; }

        public TaskStatus Status { get; set; }

        public DateTime EndTime 
        {
            get => StartTime.AddDays(Duration);
        }

        public Tasks()
        {
            StartTime = DateTime.Today;
        }
    }
}

File: client/src/app/_models/tasks.ts
import { AppUser } from "./appuser";

export interface TaskCreationDTO {
    title? : string;
    description? : string;
    startTime? : Date;
    duration? : number;
    employeeId? : number;
}

export interface Task {
    id?: number;
    title? : string;
    description? : string;
    startTime? : Date;
    endTime? : Date;
    duration? : number;
    employee? : AppUser;
    subTasks?: Array<SubTask>;
}

export interface SubTaskCreationDto {
    tasksId?: number;
    description?: string;
}

export interface SubTask {
    id?: number;
    tasksId?: number;
    description?: string;
    status?: number;
}

export interface TaskSearchDto {
    employeeId?: number;
    taskId?: number;
    status?: number;
    isOverdue?: boolean;
}